pkg/networkservice/chains/client/client.go

```diff
@@ -70,51 +74,73 @@ func WithAuthorizeClient(authorizeClient networkservice.NetworkServiceClient) Op
 //             - cc - grpc.ClientConnInterface for the endpoint to which this client should connect
-func NewClient(ctx context.Context, cc grpc.ClientConnInterface, clientOpts ...Option) networkservice.NetworkServiceClient {
+func NewClient(ctx context.Context, connectTo *url.URL, clientOpts ...Option) networkservice.NetworkServiceClient {
 	var rv networkservice.NetworkServiceClient
 	var opts = &clientOptions{
 		name:            "client-" + uuid.New().String(),
 		authorizeClient: null.NewClient(),
+		dialTimeout:     100 * time.Millisecond,
 	}
 	for _, opt := range clientOpts {
 		opt(opts)
 	}
 	rv = chain.NewNetworkServiceClient(
-		append(
-			append([]networkservice.NetworkServiceClient{
-				updatepath.NewClient(opts.name),
-				serialize.NewClient(),
-				heal.NewClient(ctx, networkservice.NewMonitorConnectionClient(cc)),
-				refresh.NewClient(ctx),
-				metadata.NewClient(),
-			}, opts.additionalFunctionality...),
+		append([]networkservice.NetworkServiceClient{
+			updatepath.NewClient(opts.name),
+			serialize.NewClient(),
+			refresh.NewClient(ctx),
+			metadata.NewClient(),
+		},
 			opts.authorizeClient,
-			networkservice.NewNetworkServiceClient(cc),
+			connect.NewNetworkServiceClient(ctx, connectTo,
+				connect.WithDialOptions(opts.dialOptions...),
+				connect.WithAdditionalFunctionality(opts.additionalFunctionality...),
+				connect.WithDialTimeout(opts.dialTimeout),
+			),
 		)...)
 	return rv
 }
 
-// Factory creates a networkservice.NetworkServiceClient by passed context.Cotnext and grpc.ClientConnInterface
-type Factory = func(ctx context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient
-
-// NewCrossConnectClientFactory - returns a (2.) case func(cc grpc.ClientConnInterface) NSM client factory.
-func NewCrossConnectClientFactory(clientOpts ...Option) Factory {
-	return func(ctx context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient {
-		return chain.NewNetworkServiceClient(
-			mechanismtranslation.NewClient(),
-			NewClient(ctx, cc, clientOpts...),
-		)
-	}
-}
-
 // NewClientFactory - returns a (3.) case func(cc grpc.ClientConnInterface) NSM client factory.
-func NewClientFactory(clientOpts ...Option) Factory {
+func NewClientFactory(clientOpts ...Option) connect.ClientFactory {
 	return func(ctx context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient {
-		return chain.NewNetworkServiceClient(
-			NewClient(ctx, cc, clientOpts...),
-		)
+		var rv networkservice.NetworkServiceClient
+		var opts = &clientOptions{
+			name:            "client-" + uuid.New().String(),
+			authorizeClient: null.NewClient(),
+		}
+		for _, opt := range clientOpts {
+			opt(opts)
+		}
+		rv = chain.NewNetworkServiceClient(
+			append(
+				append([]networkservice.NetworkServiceClient{
+					updatepath.NewClient(opts.name),
+					serialize.NewClient(),
+					heal.NewClient(ctx, networkservice.NewMonitorConnectionClient(cc)),
+					refresh.NewClient(ctx),
```

Comment:
It would be better to use first `refresh` and then `heal` chain elements:
```go
...
refresh.NewClient(ctx),
heal.NewClient(ctx, networkservice.NewMonitorConnectionClient(cc)),
...
```

Reply:
Probably even:
```go
...
refresh.NewClient(ctx),
metadata.NewClient(),
heal.NewClient(ctx, networkservice.NewMonitorConnectionClient(cc)),
...
```

Reply:
Done